src/main/java/org/opensearch/ad/rest/handler/ModelValidationActionHandler.java

```diff
@@ -0,0 +1,707 @@
+        BoolQueryBuilder query = QueryBuilders.boolQuery().filter(anomalyDetector.getFilterQuery());
+        for (Map.Entry<String, Object> entry : topEntity.entrySet()) {
+            query.filter(QueryBuilders.termQuery(entry.getKey(), entry.getValue()));
+        }
+        AggregationBuilder aggregation = getBucketAggregation(latestTime);
+        SearchSourceBuilder searchSourceBuilder = getSearchSourceBuilder(query, aggregation);
+        SearchRequest searchRequest = new SearchRequest(anomalyDetector.getIndices().toArray(new String[0])).source(searchSourceBuilder);
+        client.search(searchRequest, ActionListener.wrap(response -> processTopEntityResults(response, latestTime), listener::onFailure));
+    }
+
+    private void processTopEntityResults(SearchResponse response, long latestTime) {
+        Histogram aggregate = checkBucketResultErrors(response);
+        if (aggregate == null) {
+            return;
+        }
+        double fullBucketRate = processBucketAggregationResults(aggregate);
+        if (fullBucketRate < CONFIG_BUCKET_MINIMUM_SUCCESS_RATE) {
+            listener
+                .onFailure(
+                    new ADValidationException(
+                        CommonErrorMessages.CATEGORY_FIELD_TOO_SPARSE,
+                        DetectorValidationIssueType.CATEGORY,
+                        ValidationAspect.MODEL
+                    )
+                );
+        } else {
+            try {
+                checkFeatureQuery(latestTime);
+            } catch (Exception ex) {
+                logger.error(ex);
+                listener.onFailure(ex);
+            }
+        }
+    }
+
+    private void processFeatureQuery(SearchResponse response, long latestTime) {
+        Histogram aggregate = checkBucketResultErrors(response);
+        if (aggregate == null) {
+            return;
+        }
+
+        double fullBucketRate = processBucketAggregationResults(aggregate);
+        if (fullBucketRate < CONFIG_BUCKET_MINIMUM_SUCCESS_RATE) {
+            listener
+                .onFailure(
+                    new ADValidationException(
+                        CommonErrorMessages.FEATURE_QUERY_TOO_SPARSE,
+                        DetectorValidationIssueType.FEATURE_ATTRIBUTES,
+                        ValidationAspect.MODEL
+                    )
+                );
+            return;
+        }
+        windowDelayRecommendation(latestTime);
+    }
+
+    private void checkFeatureQuery(long latestTime) throws IOException {
+        List<String> featureFields = ParseUtils.getFeatureFieldNames(anomalyDetector, xContentRegistry);
+        AggregationBuilder aggregation = getBucketAggregation(latestTime);
+        BoolQueryBuilder query = QueryBuilders.boolQuery().filter(anomalyDetector.getFilterQuery());
+        for (String featureField : featureFields) {
+            query.filter(QueryBuilders.existsQuery(featureField));
```

Comment:
What if this is a runtime field that does not actually exist?

Reply:
I can add a check to see if `runtime_mapping` is a field in the feature query but since we still don't allow these types of queries I will prioritize other bugs and frontend first.

Reply:
sounds good